contracts/token/ERC721/extensions/ERC721Consecutive.sol

```diff
@@ -90,41 +82,27 @@ abstract contract ERC721Consecutive is IERC2309, ERC721 {
             require(to != address(0), "ERC721Consecutive: mint to the zero address");
             require(batchSize <= _maxBatchSize(), "ERC721Consecutive: batch too large");
 
-            // hook before
-            _beforeTokenTransfer(address(0), to, first, batchSize);
-
             // push an ownership checkpoint & emit event
             uint96 last = first + batchSize - 1;
             _sequentialOwnership.push(last, uint160(to));
             emit ConsecutiveTransfer(first, last, address(0), to);
 
-            // hook after
-            _afterTokenTransfer(address(0), to, first, batchSize);
+            _update(address(0), to, first, batchSize);
         }
 
         return first;
     }
 
     /**
-     * @dev See {ERC721-_mint}. Override version that restricts normal minting to after construction.
-     *
-     * Warning: Using {ERC721Consecutive} prevents using {_mint} during construction in favor of {_mintConsecutive}.
-     * After construction, {_mintConsecutive} is no longer available and {_mint} becomes available.
+     * @dev See {ERC721-_update}. Burning of tokens that have been sequentially minted must be explicit. Restricts normal minting during construction.
      */
-    function _mint(address to, uint256 tokenId) internal virtual override {
-        require(Address.isContract(address(this)), "ERC721Consecutive: can't mint during construction");
-        super._mint(to, tokenId);
-    }
-
-    /**
-     * @dev See {ERC721-_afterTokenTransfer}. Burning of tokens that have been sequentially minted must be explicit.
-     */
-    function _afterTokenTransfer(
-        address from,
-        address to,
-        uint256 firstTokenId,
-        uint256 batchSize
-    ) internal virtual override {
+    function _update(address from, address to, uint256 firstTokenId, uint256 batchSize) internal virtual override {
+        if (from == address(0) && batchSize == 1) {
+            // Using {ERC721Consecutive} prevents using {_mint} during construction in favor of {_mintConsecutive}.
+            // After construction, {_mintConsecutive} is no longer available and {_mint} becomes available.
+            require(Address.isContract(address(this)), "ERC721Consecutive: can't mint during construction");
+        }
+        super._update(from, to, firstTokenId, batchSize);
```

Comment:
Also realized that here we are not following the rule that overrides should do their own stuff before calling super (and not after). We need to add comments documenting why we are doing so.